src/util.rs

```diff
@@ -153,231 +148,13 @@ impl NetworkInterface {
     }
 }
 
-#[cfg(target_os = "linux")]
-fn sockaddr_to_network_addr(sa: *const libc::sockaddr) -> (Option<MacAddr>, Option<IpAddr>) {
-    use std::net::SocketAddr;
-
-    unsafe {
-        if sa.is_null() {
-            (None, None)
-        } else if (*sa).sa_family as libc::c_int == libc::AF_PACKET {
-            let sll: *const libc::sockaddr_ll = mem::transmute(sa);
-            let mac = MacAddr((*sll).sll_addr[0],
-                              (*sll).sll_addr[1],
-                              (*sll).sll_addr[2],
-                              (*sll).sll_addr[3],
-                              (*sll).sll_addr[4],
-                              (*sll).sll_addr[5]);
-
-            (Some(mac), None)
-        } else {
-            let addr = internal::sockaddr_to_addr(mem::transmute(sa),
-                                                  mem::size_of::<libc::sockaddr_storage>());
-
-            match addr {
-                Ok(SocketAddr::V4(sa)) => (None, Some(IpAddr::V4(*sa.ip()))),
-                Ok(SocketAddr::V6(sa)) => (None, Some(IpAddr::V6(*sa.ip()))),
-                Err(_) => (None, None),
-            }
-        }
-    }
-}
-
-#[cfg(any(target_os = "freebsd", target_os = "macos"))]
-fn sockaddr_to_network_addr(sa: *const libc::sockaddr) -> (Option<MacAddr>, Option<IpAddr>) {
-    use bindings::bpf;
-    use std::net::SocketAddr;
-
-    unsafe {
-        if sa.is_null() {
-            (None, None)
-        } else if (*sa).sa_family as libc::c_int == bpf::AF_LINK {
-            let sdl: *const bpf::sockaddr_dl = mem::transmute(sa);
-            let nlen = (*sdl).sdl_nlen as usize;
-            let mac = MacAddr((*sdl).sdl_data[nlen] as u8,
-                              (*sdl).sdl_data[nlen + 1] as u8,
-                              (*sdl).sdl_data[nlen + 2] as u8,
-                              (*sdl).sdl_data[nlen + 3] as u8,
-                              (*sdl).sdl_data[nlen + 4] as u8,
-                              (*sdl).sdl_data[nlen + 5] as u8);
-
-            (Some(mac), None)
-        } else {
-            let addr = internal::sockaddr_to_addr(mem::transmute(sa),
-                                                  mem::size_of::<libc::sockaddr_storage>());
-
-            match addr {
-                Ok(SocketAddr::V4(sa)) => (None, Some(IpAddr::V4(*sa.ip()))),
-                Ok(SocketAddr::V6(sa)) => (None, Some(IpAddr::V6(*sa.ip()))),
-                Err(_) => (None, None),
-            }
-        }
-    }
-}
-
 /// Get a list of available network interfaces for the current machine.
+/// Deprecated. Instead use the implementation available for your backend.
+/// The default one is at `pnet::datalink::interfaces`.
+#[deprecated]
```

Comment:
Does `#[deprecated]` accept a message parameter? It would be nice to mention it has moved to `pnet::datalink::interfaces()` if it does.